components/src/dynamo/trtllm/request_handlers/handler_base.py

```
@@ -621,6 +623,31 @@ async def generate_locally(
+            regex_pattern = guided_decoding.get("regex")
+            choice_list = guided_decoding.get("choice")
+            grammar = guided_decoding.get("grammar")
+            
+            if json_schema:
+                # Convert JSON schema to string if it's a dict
+                schema_str = json.dumps(json_schema) if isinstance(json_schema, dict) else json_schema
+                sampling_params.guided_decoding = GuidedDecodingParams(json=schema_str)
+                logging.info("Guided decoding enabled with JSON schema")
+            elif regex_pattern:
+                sampling_params.guided_decoding = GuidedDecodingParams(regex=regex_pattern)
+                logging.info(f"Guided decoding enabled with regex pattern")
```

Ruff (0.14.14)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)